skimage/measure/_boundary.py

```diff
@@ -0,0 +1,160 @@
+    The following elements are the coordinates of the neighboring pixels in
+    clockwise order.
+    Parameters
+    ----------
+    current ([y, x]): Coordinates of the current pixel
+    backtrack ([y, x]): Coordinates of the backtrack pixel
+    Returns
+    -------
+    List of coordinates of the moore neighborood pixels, or 0 if the backtrack
+    pixel is not a current pixel neighbor
+    """
+
+    operations = np.array(
+        [[-1, 0], [-1, 1], [0, 1], [1, 1], [1, 0], [1, -1], [0, -1], [-1, -1]]
+    )
+    neighbors = (current + operations).astype(int)
+
+    for i, point in enumerate(neighbors):
+        if np.all(point == backtrack):
+            # we return the sorted neighborhood
+            return np.concatenate((neighbors[i:], neighbors[:i]))
+    else:
+        raise RuntimeError("The backtrack is not on the neighborhood")
+
+
+def trace_boundary(coords):
+    """Coordinates of the region's boundary. The region must not have isolated
+    points.
+    
+    Parameters
+    ----------
+    coords : obj
+        Obtained with skimage.measure.regionprops()
+        
+    Returns
+    -------
+    boundary : 2D array
+        List of coordinates of pixels in the boundary
+        The first element is the most upper left pixel of the region.
+        The following coordinates are in clockwise order.
+    """
+
+    def starting_point():
+        """
+        Returns where we will start the search for the boundary. Current it
+        returns (max_x, max_y). That is, the basic example:
+
+            [0, 0, 0, 0, 0, 0],
+            [0, 1, 1, 1, 1, 0],
+            [0, 1, 1, 1, 1, 0],
+            [0, 0, 1, 1, 1, 0],
+            [0, 0, 0, 1, 0, 0],
+
+        Will start in the (4, 4) point because it has a region limit at
+        (1, 4) and another at (4, 3). On a t-shaped example, where the region
+        has borders in very far away points, say (1, 20) and (20, 1), I fear
+        that this method will begin its search at (20, 20), a point very
+        far from the boundary
+        """
+        return np.amax(coords, axis=0)
+
+    maxs = starting_point()
+
+    # should consider the minimum as well
+    # we want a image of minimal size
+    binary = np.zeros((maxs[0] + 2, maxs[1] + 2))
+
+    x = coords[:, 1]
+    y = coords[:, 0]
+    # Sets the original region as 1s
+    binary[tuple([y, x])] = 1
+
+    def find_boundary_start():
+
+        # initilization
+        # starting point is the most upper left point
+        idx_start = 0
+        while True:  # asserting that the starting point is not isolated
+            start = [y[idx_start], x[idx_start]]
+
+            # Focus Start is a 3x3 matrix centered around start
+            focus_start = binary[start[0] - 1 : start[0] + 2, start[1] - 1 : start[1] + 2]
+
+            # If the current pixel is not the only one set to one in its
+            # 3x3 region, we stop
+            if np.sum(focus_start) > 1:
+                break
+            idx_start += 1
```

Comment:
This seems a bit unnecessary. We require that the region doesn't contain isolated points, or - if you agree with my suggestion - that it is simply connected. In this case we are guaranteed to find at least one neighbor that is also part of the region.

An exception (edge case) is if the region only has a single pixel, in which case your current implementation will throw an error because it runs out of bounds (either because of `y[idx_start]` or further down when using `idx_start` which now is `idx_start=len(coords)`.

I would replace this with simply grabbing the coordinate with the lowest indices.